### DISCUSSION TITLE
What would be the best/fastest/most scalable way to create an interactive diagram?

### DISCUSSION THREAD

**Discussion by seriouscatsoserious** [jgraph/drawio#3506] [Q&A]
Hey all. I'm trying to use Draw.io to mimic the functionality seen here (https://roadmap.sh/backend). Put simply: 

1. I want to use Draw.io to create a "roadmap" ie a flowchart depicting a suggested sequential order of learning a given skill set
2. On clicking the TEXT in a given element of the flowchart, I want an overlay window to open up with some basic text and links as seen in the example site
3. I'd also like for the element itself (the shape containing the text) to have an ON/OFF state, that ideally I'd use in the future to measure cumulative progress on a given map. 

I'm coming from a Vanilla JS background, having only been learning for a few months. I'm aware that I could probably jump into the code itself and hard code a ton of values, event listeners, and functions, but I'm trying to figure out if there's either some functionality within Draw.io that could expedite this process for me, or if there's a much simipler way using a framework that I haven't learned yet.

**Comment by fabian-xentral** [comment]
Hey @seriouscatsoserious did you ever manage to get something similar put together using draw.io as a foundation?

Would be super interested to tap into your learnings.

Thank you & Best Regards
Fabian

**Reply by seriouscatsoserious** [reply]
I did not. LLMs were not quite as advanced at the time I wrote this either. I suspect you could make one in a few days now using a number of different methods.

**Reply by fabian-xentral** [reply]
thx for the quick reply !
